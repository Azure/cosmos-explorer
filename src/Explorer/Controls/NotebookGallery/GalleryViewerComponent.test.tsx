import React from "react";
import { shallow } from "enzyme";
import {
  GalleryViewerContainerComponent,
  GalleryViewerContainerComponentProps,
  FullWidthTabs,
  FullWidthTabsProps,
  GalleryCardsComponent,
  GalleryCardsComponentProps,
  GalleryViewerComponent,
  GalleryViewerComponentProps
} from "./GalleryViewerComponent";

describe("GalleryCardsComponent", () => {
  it("renders", () => {
    // TODO Mock this
    const props: GalleryCardsComponentProps = {
      data: [],
      userMetadata: undefined,
      onNotebookMetadataChange: () => Promise.resolve(),
      onClick: () => Promise.resolve()
    };

    const wrapper = shallow(<GalleryCardsComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("FullWidthTabs", () => {
  it("renders", () => {
    const props: FullWidthTabsProps = {
      officialSamplesContent: [],
      likedNotebooksContent: [],
      userMetadata: undefined,
      onClick: () => Promise.resolve()
    };

    const wrapper = shallow(<FullWidthTabs {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("GalleryViewerContainerComponent", () => {
  it("renders", () => {
    const props: GalleryViewerContainerComponentProps = {
      container: undefined
    };

    const wrapper = shallow(<GalleryViewerContainerComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("GalleryCardComponent", () => {
  it("renders", () => {
    const props: GalleryViewerComponentProps = {
      container: undefined,
      officialSamplesData: [],
      likedNotebookData: undefined
    };

    const wrapper = shallow(<GalleryViewerComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
