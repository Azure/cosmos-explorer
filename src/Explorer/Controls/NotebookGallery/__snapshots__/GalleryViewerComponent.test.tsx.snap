// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryViewerComponent renders 1`] = `
<div
  className="galleryContainer"
>
  <StyledPivot
    onLinkClick={[Function]}
    selectedKey="OfficialSamples"
  >
    <PivotItem
      headerText="Official samples"
      itemKey="OfficialSamples"
      key="OfficialSamples"
      style={
        {
          "marginTop": 20,
        }
      }
    >
      <Stack
        tokens={
          {
            "childrenGap": 10,
          }
        }
      >
        <Stack
          horizontal={true}
          tokens={
            {
              "childrenGap": 20,
              "padding": 10,
            }
          }
          wrap={true}
        >
          <StackItem
            grow={true}
          >
            <StyledSearchBox
              onChange={[Function]}
              placeholder="Search"
            />
          </StackItem>
          <StackItem>
            <StyledLabelBase>
              Sort by
            </StyledLabelBase>
          </StackItem>
          <StackItem
            styles={
              {
                "root": {
                  "minWidth": 200,
                },
              }
            }
          >
            <Dropdown
              onChange={[Function]}
              options={
                [
                  {
                    "key": 0,
                    "text": "Most viewed",
                  },
                  {
                    "key": 1,
                    "text": "Most downloaded",
                  },
                  {
                    "key": 3,
                    "text": "Most recent",
                  },
                  {
                    "key": 2,
                    "text": "Most favorited",
                  },
                ]
              }
              selectedKey={0}
            />
          </StackItem>
          <StackItem>
            <InfoComponent />
          </StackItem>
        </Stack>
        <StackItem>
          <StyledSpinnerBase
            size={3}
          />
        </StackItem>
      </Stack>
    </PivotItem>
  </StyledPivot>
</div>
`;
